pkgs/stdenv/generic/default.nix

```diff
@@ -54,7 +61,10 @@ let
       pos' = if pos != null then "‘" + pos.file + ":" + toString pos.line + "’" else "«unknown-file»";
 
       throwEvalHelp = unfreeOrBroken: whatIsWrong:
-        assert (unfreeOrBroken == "Unfree" || unfreeOrBroken == "Broken");
+        assert (unfreeOrBroken == "Unfree"
+                || unfreeOrBroken == "Broken"
+                || unfreeOrBroken == "AllowedLicense"
+                || unfreeOrBroken == "BlacklistedLicense");
```

Comment:
You could use

    builtins.elem x xs
        Return true if a value equal to x occurs in the list xs, and false otherwise.

to make this shorter. Not sure if it is better from a memory or speed standpoint though.